docs/proposals/config-dump.md

```diff
@@ -0,0 +1,248 @@
+
+Documentation for other releases can be found at [releases.K8s.io](http://releases.K8s.io).
+</strong>
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+
+# Overview
+
+Here we describe a proposal for retrieving configurations (configs) that govern the behavior of any one of the Kubernetes (K8s) components. By components we mean any of the following: apiserver, scheduler, control-manager, and kubelet. We expect that these configurations be for the most part static through the life of the component but fully understand that changes from specs may affect these. Consequently, any retrieval of the configurations is to be considered true and valid at the time of retrieval and we will assume that the user is well aware of any transient behavior.   
+
+We propose several important changes:
+* new kubectl command: 
+    * `kubectl get componentconfigs <componentName> (alias compcs) --id <componentId>`
+* adding new API: 
+    * `/api/alpha/component/name/<componentName>/configurations?id=<componentId>` \*
+* creating maintainable configuration objects to components
+* adding api endpoints to individual components to request configs directly, e.g., in the case of a standalone node running kubelet
+    * `/api/alpha/configurations` 
+
+\* Query parameter id will optionally allow you to select a particular component in the case where there may be a set of them, such as in the case of kubelet. Where id is left out and we have a set of components we should expect to produce a list of configs results. 
+
+## Motivation
+
+By necessity Kubernetes is a stateful thing made up of components who have been configured in a particular way. Currently the K8s code base, as 09/2016, does not support a way nor means by which to query for the set of configurations that led to the current state of any component. As a consequence it is extremely difficult to debug the current state of a K8s cluster, especially when original configurations have been lost, or are simply not available. 
+
+## Design
+
+### API changes
+New endpoints in the api are to be created: 
+
+```
+/api/alpha/component/name/<componentName>/configurations?id=<componentId>
+```
+
+supporting the following HTTP method
+
+``` 
+GET - retrieve component configurations. **
+```
+
+\*\* The `PUT` method is to be discussed in the [Future work](#Future work) section of this proposal.
+
+An interface should be created at the component level that allows different classifications of configurations to be added to a particular component. The point being that the final configuration object is composed of a multitude classes of very specific classifications presumably defined by interested parties.
+
+Example request of a cluster with one running apiserver component:
+
+```
+kubectl get componentconfigs apiserver
+```
+
+for which the api call looks like:
+
+```
+/api/alpha/component/name/apiserver/configurations
```

Comment:
There can be multiple apiservers, and multiple configurations in the cluster at any one time.